### PR TITLE
Output skipped tests from qunit to xunit & tap

### PR DESCRIPTION
**Pull request by step2yeung**
This is a copy of #606 with the additional reporting in tap reporter.
Since #606 has not been touched in 4 months, I created a new PR.

This will output skipped tests for both xunit & tap reporter.

### CODE REVIEW

**Comment by step2yeung** [comment]
cc @nickiaconis

File: lib/ci/test_reporters/tap_reporter.js [hunk 1]
```diff
@@ -19,17 +20,24 @@ TapReporter.prototype = {
...
+      '1..' + this.total
     ]
+    lines.push('# tests ' + this.total)
+    lines.push('# pass  ' + this.pass)
+    if (this.skipped > 0) {
+      lines.push('# skip  ' + this.skipped)
+    }
+    lines.push('# fail  ' + (this.total - this.pass - this.skipped))
+
     if (this.pass === this.total){
```
Comment (nickiaconis) [inline lib/ci/test_reporters/tap_reporter.js]: This will need to account for skipped tests.

File: lib/ci/test_reporters/tap_reporter.js [hunk 2]
```diff
@@ -19,17 +20,24 @@ TapReporter.prototype = {
...
-      '1..' + this.total,
-      '# tests ' + this.total,
-      '# pass  ' + this.pass,
-      '# fail  ' + (this.total - this.pass)
+      '1..' + this.total
     ]
+    lines.push('# tests ' + this.total)
+    lines.push('# pass  ' + this.pass)
+    if (this.skipped > 0) {
+      lines.push('# skip  ' + this.skipped)
```
Comment (nickiaconis) [inline lib/ci/test_reporters/tap_reporter.js]: I would recommend the number of skipped tests always be displayed, even if it is 0.

**Comment by nickiaconis** [comment]
I'm really looking forward to having this data in the reports. :+1:

**Comment by step2yeung** [comment]
ok, totally screwed up some rebasing.. will repost a PR

**Pull request closed without merging**